inference-engine/tests/functional/inference_engine/CMakeLists.txt

```diff
@@ -11,19 +11,23 @@ addIeTargetTest(
         INCLUDES
             # TODO: remove after removing `cnn_network_ngraph_imp.hpp`
             ${IE_MAIN_SOURCE_DIR}/src/inference_engine
+            ${OpenVINO_MAIN_SOURCE_DIR}/docs/template_extension
         EXCLUDED_SOURCE_DIRS
             ${CMAKE_CURRENT_SOURCE_DIR}/extension_lib
         LINK_LIBRARIES
             gmock
             funcTestUtils
             ngraphFunctions
             inference_engine_transformations
+            template_extension
```

Comment:
If I remember right functional tests load the extension library in the runtime, it is why they are not linked with this library.